File: src/main/java/com/graqr/threshr/model/redsky/products/FindsStory.java
package com.graqr.threshr.model.redsky.products;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record FindsStory(
        String id, @JsonProperty("channel_id") String channelId, String description,
        @JsonProperty("display_date") String displayDate,
        @JsonProperty("highlight_color") String highlightColor, Image image, List<Image> images,
        String label, String name, @JsonProperty("short_title") String shortTitle, String title,
        String url) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/FindsStory.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class FindsStory(
    val id: String,
    @param:JsonProperty("channel_id") val channelId: String,
    val description: String,
    @param:JsonProperty("display_date") val displayDate: String,
    @param:JsonProperty("highlight_color") val highlightColor: String,
    val image: Image,
    val images: List<Image>,
    val label: String,
    val name: String,
    @param:JsonProperty("short_title") val shortTitle: String,
    val title: String,
    val url: String
) {}
